src/environmental/water_quality/surface_water_quality.py

```diff
@@ -0,0 +1,304 @@
+    return water_quality_not_in_db
+
+
+def store_surface_water_quality_to_db(engine: Engine, catchment_area: gpd.GeoDataFrame) -> None:
+    """
+    Fetch surface water quality data from ECAN for the specified catchment area and store it in the database.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    catchment_area : gpd.GeoDataFrame
+        A GeoDataFrame representing the catchment area.
+    """
+    # Define the table name for storing the surface water quality data
+    table_name = "surface_water_quality"
+    # Check if the table already exists in the database
+    if check_table_exists(engine, table_name):
+        # Retrieve surface water quality data from ECAN for the requested catchment area
+        # that is not already present in the existing database
+        water_quality_not_in_db = get_surface_water_quality_not_in_db(engine, catchment_area)
+        # Check if any new surface water quality data was retrieved
+        if water_quality_not_in_db.empty:
+            # If no new data is found, log a message indicating that no new data was found
+            log.info("No new surface water quality data found for the requested catchment area.")
+        else:
+            # If new data is found, add it to the relevant table in the database
+            log.info(f"Adding '{table_name}' for sites within the requested catchment area to the database.")
+            water_quality_not_in_db.to_postgis(table_name, engine, index=False, if_exists="append")
+            log.info(f"Successfully added '{table_name}' to the database.")
+    else:
+        # Fetch surface water quality data from ECAN for the requested catchment area
+        surface_water_quality = get_surface_water_quality_with_geom(engine, catchment_area)
+        # Check if any surface water quality data was fetched
+        if not surface_water_quality.empty:
+            # If data is found, add it to the relevant table in the database
+            log.info(f"Adding '{table_name}' for sites within the requested catchment area to the database.")
+            surface_water_quality.to_postgis(table_name, engine, index=False, if_exists="replace")
+            log.info(f"Successfully added '{table_name}' to the database.")
+
+
+def get_req_surface_water_quality_from_db(
+        engine: Engine,
+        catchment_area: gpd.GeoDataFrame,
+        year: Optional[int] = None) -> gpd.GeoDataFrame:
+    """
+    Retrieve the latest requested surface water quality data from the database for the specified catchment area.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    catchment_area : gpd.GeoDataFrame
+        A GeoDataFrame representing the catchment area.
+    year : Optional[int] = None
+        The year from which to include the latest surface water quality data for sites within the specified
```

Comment:
The implementation looks like a minimum year filter but I found that hard to immediately understand from the docstring

Reply:
Regarding the function `get_req_surface_water_quality_from_db()`, I have included it in response to your inquiry about how to get the latest data for each site. Please note that it still depends on front-end visualization preferences, so it will likely require further adjustments in the future. I am sharing it here as an example and a starting point for now.

Reply:
As requested, this function has been removed.